### PR TITLE
Fixing broken tests

### PR DESCRIPTION
**Pull request by kessler-frost**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

* Fixing tests that are breaking in develop due to some recent changes.
* Some functional tests are also misplaced into unit tests so those will be moved to their appropriate directories.
* Some unit tests are hitting the live DB so those will be changed to hit an in memory test DB.

### CODE REVIEW

**Comment by kessler-frost** [comment]
Also unskipped and fixed some tests hence the slight increase in coverage.

File: tests/functional_tests/_core/execution_test.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    """Note: This is not a full unit test for the _run_task method. Rather, this is intended to test the diff introduced to write the sublattice electron id in the Database."""
+
+    # class MockResult:
+    #     dispatch_id = "test"
+
+    # def mock_func():
+    #     return MockResult()
+    # class MockSerializedCallable:
+    #     def get_deserialized(self):
+    #         return mock_func
```
Comment (FyzHsn) [inline tests/functional_tests/_core/execution_test.py:72]: (non-blocking) Can we delete these commented lines?

**Review by FyzHsn — APPROVED** [review]
You get an A+ for mocking and testing! I added one minor non-bocking comment. Up the code coverage and this PR is good ter go!

File: tests/functional_tests/_core/execution_test.py (hunk 1 above)
Comment (FyzHsn) [reply tests/functional_tests/_core/execution_test.py:72]: You need to test that one line that you added

Comment (kessler-frost) [reply tests/functional_tests/_core/execution_test.py:72]: On it.

**Review by santoshkumarradha — COMMENTED** [review]
Approved ✅

**Pull request merged**